dnf/cli/cli.py

```diff
@@ -737,9 +737,13 @@ def _process_demands(self):
         demands = self.demands
         repos = self.base.repos
 
-        if demands.root_user:
+        if demands.root_user or demands.root_writable:
             if not os.getegid() == 0:
                 raise dnf.exceptions.Error(_('This command has to be run under the root user.'))
+        # Are we processing a write command but without /usr writable?
+        if demands.root_writable and self.base.conf.installroot == '/':
```

Comment:
I think this can be made more generic and say that if the installroot is not writable, we can't do anything.

Reply:
I thought about that, but the goal here is to make the "default" case nicer - we're trying to help the case of people who read `yum install foo` on some website.  Anyone using `--installroot` should generally know what they're doing.  That said I'm fine adding a case for the "non-/" installroot case too with a more general message.

Reply:
I think if you're going to do this for `/`, you should support it also for `--installroot`.